### PR TITLE
Add last-update-time to the content items of an organisation

### PR DESCRIPTION
**Pull request by ikennaokpala**
[Trello card](https://trello.com/c/7swQPTrR/63-story-add-last-update-time-to-the-content-items-of-an-organisation)

## Description 
For every content item, content designers want to see the last time a
content item was updated for an organisation.

Here a new column/attribute (i.e `last_updated_at`) has been added to the content_item(s).

The `last_updated_at` attribute’s value is obtained via from the publishing api [1] via
the gds-api-adapters gem.

The gds-api-adapters gem provides access to the publishing_api endpoints

For obtaining the last time a piece of content_item was updated, this endpoint [1] is 
queried with the content_id supplied.

All organisation importer tests have been affected and a before hook introduced to
stub calls to get_content.

***NB:***

Kindly take note that publishing api can only be access via the dev VM [2]. 

[1] /v2/content/:content_id
[2] http://publishing-api.dev.gov.uk/v2/content/:content_id

### CODE REVIEW

**Comment by pmanrubia** [comment]
@ikennaokpala according to the [acceptance criteria of the Trello card](https://trello.com/c/7swQPTrR/63-story-add-last-update-time-to-the-content-items-of-an-organisation), it seems we also need to display the value in the table.

**Comment by ikennaokpala** [comment]
@pmanrubia thanks for letting me know

**Comment by ikennaokpala** [comment]
@pmanrubia I have added the view code for this PR. You will notice that in #14 I made changes to the css selectors that also affected the narrative of the view tests. that will be properly reflected here once #14 is merged.. This will then need to be rebased

**Comment by ikennaokpala** [comment]
Moving this back to in progress to refactor and retrieve last updated at from the content item store.

File: app/models/importers/organisation.rb [hunk 1]
```diff
@@ -13,8 +13,14 @@ def run
     loop do
       result = search_content_items_for_organisation
       result.each do |content_item_attributes|
-        attributes = content_item_attributes.slice('content_id')
-        organisation.content_items << ContentItem.new(attributes)
+        content_id = content_item_attributes['content_id']
+        link = content_item_attributes['link']
+
+        if content_id
```
Comment (pmanrubia) [inline app/models/importers/organisation.rb]: Lines 20 and 21 feels to be doing too much. It is not very easy to read

I would go with something similar to:

```ruby
        if content_id.present?
          link = content_item_attributes['link']
          attributes = content_item_attributes.slice('content_id')
          attributes.merge!(last_updated_at: find_public_updated_at(link))

          organisation.content_items << ContentItem.new(attributes)
        else
          Logger.warn('There is not content_id for ', slug)
        end
```

Although, in incoming commits we will extract a lot of logic from this class

File: db/migrate/20161130153206_add_last_updated_at_to_content_items.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddLastUpdatedAtToContentItems < ActiveRecord::Migration[5.0]
+  def change
+    add_column :content_items, :last_updated_at, :datetime
```
Comment (pmanrubia) [inline db/migrate/20161130153206_add_last_updated_at_to_content_items.rb]: I would mimic the same name as in the `content_store`. This will improve code readability.
I would move from `last_updated_at` to `public_updated_at`.

File: spec/models/importers/organisation_spec.rb [hunk 3]
```diff
@@ -1,19 +1,40 @@
 require 'rails_helper'
```
Comment (pmanrubia) [inline spec/models/importers/organisation_spec.rb]: There are a lot of things hapening in this commit.
Could you please extract separate commits for your refactorings. It is going to be much easier to review.

File: db/migrate/20161130153206_add_last_updated_at_to_content_items.rb (hunk 2 above)
Comment (ikennaokpala) [inline db/migrate/20161130153206_add_last_updated_at_to_content_items.rb]: This `last_updated_at` is as discussed based on the initial plan to use publishing api..

Comment (pmanrubia) [inline db/migrate/20161130153206_add_last_updated_at_to_content_items.rb]: Yes, that was part of the story. But as part of the story, if we find out that the field name is `public_updated_at` it might be good to mimic in our models. In general, the code will be more readable.

Comment (ikennaokpala) [inline db/migrate/20161130153206_add_last_updated_at_to_content_items.rb]: @pmanrubia that is fine.. but then remember when I talked about context..  these names should at most be relattive to the context..  will push an update

File: spec/models/importers/organisation_spec.rb (hunk 3 above)
Comment (pmanrubia) [inline spec/models/importers/organisation_spec.rb]: I would like to see a commit where you add your test, and then another commit where you do the Dry-up. Something similar to what you did in 4dcd67a2e3237bb53cb030a8ad06a29fbe9ebb79

File: db/migrate/20161130153206_add_last_updated_at_to_content_items.rb (hunk 2 above)
Comment (pmanrubia) [inline db/migrate/20161130153206_add_last_updated_at_to_content_items.rb]: @ikennaokpala I don't get what you mean by `relative to the context`. You prefer to stick to `last_updated_at`. What is your preference? You don't have to push an update if you don't agree.

Comment (ikennaokpala) [inline db/migrate/20161130153206_add_last_updated_at_to_content_items.rb]: I am not sure `public_updated_at` fits our current context.. I am happy that if this turns out wrong rails make is easy to update this..

File: spec/models/importers/organisation_spec.rb (hunk 3 above)
Comment (ikennaokpala) [inline spec/models/importers/organisation_spec.rb]: that is fine.. will update.. I thought it clear enough.. cheers for point that out

**Comment by pmanrubia** [comment]
@ikennaokpala I have prepared a commit with small updates. If you are happy with them please fix up the commit and merge.
The test is not looking very good because we are just doing too much in that class. We will refactor it in incoming stories once we have a clearer picture of the moving parts to import an organisation.

**Comment by ikennaokpala** [comment]
@pmanrubia sorry I have resolved the merge ocnflicts..  kindly update with your fixup.
